ga4gh/backend.py

```diff
@@ -496,15 +496,10 @@ def variantAnnotationSetsGenerator(self, request):
         """
         compoundId = datamodel.VariantSetCompoundId.parse(request.variantSetId)
         dataset = self.getDataRepository().getDataset(compoundId.datasetId)
-        results = []
-        for annset in dataset.getVariantAnnotationSets():
-            try:
-                variantSetId = request.variantSetId
-            except ValueError:
-                variantSetId = ""
-            if str(annset._variantSetId) == str(variantSetId):
-                results.append(annset)
-        return self._objectListGenerator(request, results)
+        variantSet = dataset.getVariantSet(request.variantSetId)
```

Comment:
I'm pretty sure this isn't correct to the schemas since the annotation set and the variant set are not necessarily in the same dataset. Although the implementation does place them in the same dataset currently, this was due to being backed by files. Now we shouldn't have to make the same assumptions when relating variant and annotation sets.

I would discourage writing code that solidifies the assumption. The SQL repo gives us a nice path for setting foreign keys, we can take advantage of it by removing the relational assumptions from our python classes. This makes separation of concerns much more clear and allows each protocol to interact with the other's via a simple API.

To fulfill a searchVariantAnnotationSets request we will find all annotationSets that have the given variantSetId, as opposed to the variantSet that has had these annotation sets added to them. This means we only look at one table when responding to the request and can handle meta-dataset searches properly. The SQL backing moves us towards better satisfying the schemas.

In general, I believe we are complicating our picture by maintaining the class-based container model. Being able to reference objects directly in a table will allow us to remove this complexity. Am I wrong to assume `datasets.getVariantSet` will be translated to SQL call? I leave this message here in hopes that we will maintain a single data structure in SQL as opposed to a tree of classes, and I know that some of these are stop-gap measures.

I realize some of the changes to the variants datamodel needed to be made to add in the values that were being populated by the metadata JSON. Could you provide justification for the other refactors, could they be moved into a separate PR? What do we gain?